lib/src/Uncertainty/Algorithm/Simulation/StandardSpaceCrossEntropyImportanceSampling.cxx

```diff
@@ -0,0 +1,156 @@
+  : CrossEntropyImportanceSampling(event,rhoQuantile)
+  {
+
+  initialDistribution_ = Normal(initialDistribution_.getDimension());
+
+  auxiliaryDistribution_ = Normal(initialDistribution_.getDimension());
+  rhoQuantile_ = (event.getOperator()(0, 1) ? rhoQuantile : 1.0 - rhoQuantile);
+}
+
+// Compute Output Samples
+Sample StandardSpaceCrossEntropyImportanceSampling::computeOutputSamples(const Sample & inputSamples) const
+{
+
+  Sample outputSamples = getEvent().getFunction()(getEvent().getAntecedent().getDistribution().getInverseIsoProbabilisticTransformation()(inputSamples));
+
+  return outputSamples;
+}
+
+// Update auxiliary distribution
+void StandardSpaceCrossEntropyImportanceSampling::updateAuxiliaryDistribution(const Point & auxiliaryDistributionParameters)
+{
+
+  Point temporaryParameters = auxiliaryDistribution_.getParameter();
+  
+  for (UnsignedInteger i = 0; i < auxiliaryDistributionParameters.getDimension(); ++i)
+  {
+    temporaryParameters[i] = auxiliaryDistributionParameters[i];
+  }
+  auxiliaryDistribution_.setParameter(temporaryParameters);
+} 
+  
+// Optimize auxiliary distribution parameters
+Point StandardSpaceCrossEntropyImportanceSampling::optimizeAuxiliaryDistributionParameters(const Sample & auxiliaryCriticInputSamples) const
+{
+  //evaluate initial PDF on failure auxiliaryInputSamples
+  Point criticSamplesInitialPDFValue   = initialDistribution_.computePDF(auxiliaryCriticInputSamples).asPoint(); 
+  //evaluate auxiliary PDF on failure auxiliaryInputSamples
+  Point criticSamplesAuxiliaryPDFValue = auxiliaryDistribution_.computePDF(auxiliaryCriticInputSamples).asPoint(); 
+
+  //evaluate initial PDF on failure auxiliaryInputSamples
+  Point criticSamplesInitialLogPDFValue   = initialDistribution_.computeLogPDF(auxiliaryCriticInputSamples).asPoint();
+  //evaluate auxiliary PDF on failure auxiliaryInputSamples 
+  Point criticSamplesAuxiliaryLogPDFValue = auxiliaryDistribution_.computeLogPDF(auxiliaryCriticInputSamples).asPoint();
+  
+  // calculation of denominator of estimators
+  Scalar sumPdfCritic = 0.0;
+  Point PDFRatio(auxiliaryCriticInputSamples.getSize()) ;
+  
+  for(UnsignedInteger i = 0; i < auxiliaryCriticInputSamples.getSize(); ++i)
+  {
+    PDFRatio[i] = std::exp(criticSamplesInitialLogPDFValue[i] - criticSamplesAuxiliaryLogPDFValue[i]);
+    sumPdfCritic += PDFRatio[i] ;
+  }
+  
+  // Calculation of updated mean of auxiliary distribution
+  Point mean(auxiliaryCriticInputSamples.getDimension());
+  
+  for(UnsignedInteger i = 0; i < auxiliaryCriticInputSamples.getDimension(); ++i)
+  {
+  
+    Scalar numeratorMeanCalculation = 0.0;
+  
+      for(UnsignedInteger j = 0; j < auxiliaryCriticInputSamples.getSize(); ++j)
+      {
+      numeratorMeanCalculation += std::exp(criticSamplesInitialLogPDFValue[j] - criticSamplesAuxiliaryLogPDFValue[j])* auxiliaryCriticInputSamples(j,i);
+      }
+    
+    mean[i]= numeratorMeanCalculation / sumPdfCritic; 
```

Comment:
```suggestion
    mean[i] = numeratorMeanCalculation / sumPdfCritic; 
```
What if `sumPdfCritic` is zero?

Reply:
This should note happen. Nevertheless, an exception has been added.